geo/src/algorithm/remove_repeated_points.rs

```diff
@@ -0,0 +1,395 @@
+use crate::{
+    CoordNum, Geometry, Line, LineString, MultiLineString, MultiPoint, MultiPolygon, Point,
+    Polygon, Rect, Triangle,
+};
+use geo_types::GeometryCollection;
+use num_traits::FromPrimitive;
+
+/// Remove repeated points from a `MultiPoint` and repeated consecutive coordinates
+/// from `LineString`, `Polygon`, `MultiLineString` and `MultiPolygon`.
+///
+/// For `GeometryCollection` it individually removes the repeated points
+/// of each geometry in the collection.
+///
+/// For `Point`, `Line`, `Rect` and `Triangle` it returns a clone of the geometry.
+pub trait RemoveRepeatedPoints<T>
+where
+    T: CoordNum + FromPrimitive,
+{
+    /// Create a new geometry with (consecutive) repeated points removed.
+    fn remove_repeated_points(&self) -> Self;
+}
+
+impl<T> RemoveRepeatedPoints<T> for MultiPoint<T>
+where
+    T: CoordNum + FromPrimitive,
+{
+    /// Create a MultiPoint with repeated points removed.
+    fn remove_repeated_points(&self) -> Self {
+        let mut points = vec![];
+        for p in self.0.iter() {
+            if !points.contains(p) {
+                points.push(*p);
+            }
+        }
+        MultiPoint(points)
+    }
+}
+
+impl<T> RemoveRepeatedPoints<T> for LineString<T>
+where
+    T: CoordNum + FromPrimitive,
+{
+    /// Create a LineString with consecutive repeated points removed.
+    fn remove_repeated_points(&self) -> Self {
+        let mut coords = vec![];
+        for c in self.0.iter() {
```

Comment:
This is the same as `LineString(self.0.dedup().collect())` right?  If so, can you use that instead?

Reply:
Oh for sure, I had completely missed the `dedup` method of Vec !

Reply:
Done.

But because `dedup` acts in-place I couldn't implement it in one line as proposed and I had to clone the coordinate vector first.

However, this raises the question of whether `RemoveRepeatedPoints` should create a new geometry (which is proposed here) or act inplace (which would not be a problem from a conceptual point of view in my opinion, and which should not be much more complicated to implement).

Reply:
>  because dedup acts in-place I couldn't implement it in one line as proposed and I had to clone the coordinate vector first.

Oh sorry, my mistake.

> whether RemoveRepeatedPoints should create a new geometry 

I think ideally we would have both! This is a common pattern in geo, though we're a bit inconsistent in how naming goes. I'd suggest:

```
remove_repeated_points
remove_repeated_points_mut
```